x-pack/test/api_integration/apis/actions/fire.ts

```diff
@@ -193,5 +193,41 @@ export default function({ getService }: KibanaFunctionalTestDefaultProviders) {
         source: 'action:test.index-record',
       });
     });
+
+    it(`should return 404 when action doesn't exist`, async () => {
+      await supertest
+        .post('/api/action/1/_fire')
+        .set('kbn-xsrf', 'foo')
+        .send({
+          params: { foo: true },
+        })
+        .expect(404)
+        .then((resp: any) => {
```

Comment:
__note:__ that's kind of funny that we use both `await` and `then` :) Do you think it's unnecessary to split assertion like in example below?

```ts
const { body: response } = await supertest
        .post('/api/action/1/_fire')
        .set('kbn-xsrf', 'foo')
        .send({ params: { foo: true } })
        .expect(404);

expect(response).to.eql({
  statusCode: 404,
  error: 'Not Found',
  message: 'Saved object [action/1] not found',
});
```